### PR TITLE
fix(schemas): set default password for tenant (#4535)

### LINKED ISSUE #4535: bug: Unable to seed database on Neon
### Describe the bug
I'm trying to seed the Logto database on [neon.tech](https://neon.tech/)'s managed Postgres platform, but that is not possible, due to a empty password on a role, I think gets created in the seed phase.

<pre>
> cli
> logto db seed

[info] ✔ Create tables
info [default] Generated config oidc.privateKeys
info [default] Generated config oidc.cookieKeys
info ✔ [default] Seed OIDC config
info [admin] Generated config oidc.privateKeys
info [admin] Generated config oidc.cookieKeys
info ✔ [admin] Seed OIDC config
info ✔ Cloud Service Application successfully created for: default
info ✔ Seed data

error error: Received HTTP code 400 from control plane: {"error":"Neon does not support roles with empty passwords"}
  at Parser.parseErrorMessage (/etc/logto/node_modules/.pnpm/pg-protocol@1.6.0/node_modules/pg-protocol/dist/parser.js:287:98)
  at Parser.handlePacket (/etc/logto/node_modules/.pnpm/pg-protocol@1.6.0/node_modules/pg-protocol/dist/parser.js:126:29)
  at Parser.parse (/etc/logto/node_modules/.pnpm/pg-protocol@1.6.0/node_modules/pg-protocol/dist/parser.js:39:38)
  at TLSSocket.<anonymous> (/etc/logto/node_modules/.pnpm/pg-protocol@1.6.0/node_modules/pg-protocol/dist/index.js:11:42)
  at TLSSocket.emit (node:events:514:28)
  at addChunk (node:internal/streams/readable:324:12)
  at readableAddChunk (node:internal/streams/readable:297:9)
  at Readable.push (node:internal/streams/readable:234:10)
  at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) {
    length: 189,
    severity: 'ERROR',
    code: 'XX000',
    detail: undefined,
    hint: undefined,
    position: undefined,
    internalPosition: undefined,
    internalQuery: undefined,
    where: undefined,
    schema: undefined,
    table: undefined,
    column: undefined,
    dataType: undefined,
    constraint: undefined,
    file: 'control_plane_connector.c',
    line: '276',
    routine: 'SendDeltasToControlPlane'
  }
</pre>

My connection string looks like this:

<pre>
postgres://[username]:[password]@ep-crimson-flower-88570255.eu-central-1.aws.neon.tech/logto?sslmode=require
</pre>

### Expected behavior
Able to seed the database and test it.

### How to reproduce?

1. Create a new user: https://console.neon.tech
2. Create new project with version 15 of Postgres
3. Copy the connection string from the neon.tech UI
4. Set `DB_URL` in [`docker-compose.yml`](https://github.com/logto-io/logto/blob/master/docker-compose.yml)
5. Add `?sslmode=require` in end of the connection string
6. Run `docker compose up` and wait for the seed error

### Context

- [x] Self-hosted, Logto version = 1.8.0
  - [x] Container (Docker image)

### PR DESCRIPTION
**Pull request by 5aaee9**
## Summary

This PR fixed #4535, neon is able to finish seeding now.

## Testing

Tested on local machine.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/cli/src/commands/database/seed/tables.ts [hunk 1]
```diff
@@ -118,8 +119,9 @@ export const createTables = async (connection: DatabaseTransactionConnection) =>
   ];
   const sorted = allQueries.slice().sort(compareQuery);
   const database = await getDatabaseName(connection, true);
+  const password = generateStandardId(32);
```
Comment (xiaoyijun) [inline packages/cli/src/commands/database/seed/tables.ts:122]: Is it a good practice to create a random string for the role password?
May be we need to get the password from the seed command.
@charIeszhao @gao-sun what do you think?

Comment (simeng-li) [reply packages/cli/src/commands/database/seed/tables.ts:122]: Also, this feature better be optional.

Comment (gao-sun) [reply packages/cli/src/commands/database/seed/tables.ts:122]: i think it's fine. this role is a base role with no login. the only concern is if we should store the password somewhere, or print it after seed

Comment (charIeszhao) [reply packages/cli/src/commands/database/seed/tables.ts:122]: Maybe we should add a new option in the db seed CLI command to generate this random password, and print it out in console if the command is executed with the option.

Comment (5aaee9) [reply packages/cli/src/commands/database/seed/tables.ts]: Updated, now use `--encrypt-base-role` to create base role with password

File: packages/schemas/tables/_before_all.sql [hunk 2]
```diff
@@ -1,3 +1,3 @@
 /* This SQL will run before all other queries. */
 
-create role logto_tenant_${database} noinherit;
+create role logto_tenant_${database} password '${password}' noinherit;
```
Comment (darcyYe) [inline packages/schemas/tables/_before_all.sql:3]: nit: `create role ${role_name} password '' noinherit;` could sometimes lead to PostgreSQL warning, better keep the query simple (e.g. `create role ${role_name} noinherit;`) when the `--encrypt-base-role` flag is false.

**Review by darcyYe — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

File: packages/cli/src/commands/database/seed/index.ts [hunk 3]
```diff
@@ -21,9 +26,14 @@ export const seedByPool = async (pool: DatabasePool, cloud = false, test = false
       );
     }
 
-    await oraPromise(createTables(connection), {
+    const tableInfo = await oraPromise(createTables(connection, encryptBaseRole), {
       text: 'Create tables',
     });
+
+    if (tableInfo.password.length > 0) {
+      consoleLog.info('base role password:', tableInfo.password);
```
Comment (simeng-li) [inline packages/cli/src/commands/database/seed/index.ts:34]: better avoid logging plain text passwords

**Review by simeng-li — APPROVED** [review]
overall LGTM

**Comment by wangsijie** [comment]
@5aaee9 One last problem to fix: can you sign your commits in GPG?

**Pull request merged**